sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/DeclarativeConfigurationCustomizer.java

```diff
@@ -8,7 +8,14 @@
 import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfigurationModel;
 import java.util.function.Function;
 
+/** A service provider interface (SPI) for customizing declarative configuration. */
 public interface DeclarativeConfigurationCustomizer {
+  /**
+   * Method invoked when configuring the SDK to allow further customization of the declarative
+   * configuration.
+   *
+   * @param customizer the customizer to add
+   */
   void addModelCustomizer(
       Function<OpenTelemetryConfigurationModel, OpenTelemetryConfigurationModel> customizer);
```

Comment:
Just realizing that this is going to be a problematic contract from a packaging standpoint..

- Currently, the SPI lives in `opentelemetry-sdk-extension-incubator`
- In the future, we'll want to move it to `opentelemetry-sdk-extension-autoconfigure-spi` with the rest of the SPIs
- That leaves a thorny question for where `OpenTelemetryConfigurationModel` and the rest of the data model lives. First guess would be `opentelemetry-sdk-extension-autoconfigure`, but that creates a dependency cycle. They could live in `opentelemetry-sdk-extension-autoconfigure-spi`, or `opentelemetry-sdk`, or `opentelemetry-sdk-common`, or some new `opentelemetry-sdk-config` package. 

Idk, let's cross that bridge when we get there.